build-script.py

```diff
@@ -505,36 +562,27 @@ def verify_c_syntax_nodes_match():
 
 
 def run_tests(
-    toolchain, build_dir, multiroot_data_file, release, filecheck_exec,
-    skip_lit_tests, verbose
+    toolchain: str, build_dir: Optional[str], multiroot_data_file: Optional[str],
+    release: bool, filecheck_exec: Optional[str], skip_lit_tests: bool, verbose: bool
 ):
     print("** Running SwiftSyntax Tests **")
 
-    if skip_lit_tests:
-        lit_success = True
-    else:
-        lit_success = run_lit_tests(
+    if not skip_lit_tests:
+        run_lit_tests(
```

Comment:
I know this is the current behavior but why we don't run xctest cases when lit-tests fails? I feel we should run them both unconditionally, and raise a failure when either of them fails.

Reply:
I think there are three reasons:
1. I like the way build-script fails early. That way the log output of lit-based tests are closer to the bottom (and thus easier to read) than when we continue to execute xctest commands
2. It follows the current paradigm of the build-script that if one command fails it shouldn’t execute any other commands
3. It’s slightly easier to implement this way

If you have strong opinions on this, we can change it but I like it the way it is.

Reply:
I see. I agree that it's nice we can see the failures closer to the bottom. I still think it's also nice too all failures for both test suites, but it's not a strong opinion, and I can skip lit tests if I want. So let's keep it as it is.